flow/record/base.py

```diff
@@ -82,27 +108,31 @@ def _unpack(__cls, {args}):
 """
 
 
-class Peekable:
+class Peekable(BinaryIO):
     """Wrapper class for adding .peek() to a file object."""
 
     def __init__(self, fd):
         self.fd = fd
         self.buffer = None
+        self.peeked_data = None
 
     def peek(self, size):
         if self.buffer is not None:
             raise BufferError("Only 1 peek allowed")
         data = self.fd.read(size)
         self.buffer = io.BytesIO(data)
+        self.peeked_data = data
```

Comment:
What is this needed for btw?

Reply:
You were right, this wasn't needed as we could use the buffer property.